src/lib/dialog/dialog.ts

```diff
@@ -228,7 +228,9 @@ export class MdDialog {
     }
 
     if (componentOrTemplateRef instanceof TemplateRef) {
-      dialogContainer.attachTemplatePortal(new TemplatePortal(componentOrTemplateRef, null!));
+      dialogContainer.attachTemplatePortal(
+        new TemplatePortal<T>(componentOrTemplateRef, null!,
```

Comment:
This was the logic before I made the change so I left it as is.

I think if a change is required here it should be handled in a different PR not to mixup things.

It does work...